### PR TITLE
OEP: Capability Options

### PR DESCRIPTION
**Pull request by joshmeranda**
Add OEP doc for capability options

### CODE REVIEW

File: enhancements/gateway/20230309-customize-capabilty-install.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+enum NodeDriver {
+  Unknown = 0;
+  None = 1;
+  ExternalPromDriver = 2;
+}
+
+message MetricsOptions {
+  NodeDriver nodeDriver = 1;
+}
+```
```
Comment (kralicky) [inline enhancements/gateway/20230309-customize-capabilty-install.md:40]: I know this is just an example, but for the real implementation I would suggest the following naming:
```proto
syntax = "proto3";
option go_package = "github.com/rancher/opni/plugins/metrics/pkg/apis/node";

import "google/protobuf/empty.proto";

service NodeConfiguration {
  rpc Configure(MetricsOptions) returns (google.protobuf.Empty);
}

enum NodeDriver {
  Unknown = 0;
  None = 1;
  ExternalPromDriver = 2;
}

message MetricsOptions {
  NodeDriver nodeDriver = 1;
}
```

File: enhancements/gateway/20230309-customize-capabilty-install.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
+# Title: Customize Capability Install
+
+## Summary:
+When installing capabilities to agent's, Opni should allow for users to configure some capability options such as specifying node drivers and enabling / disabling capability features. This would make Opni agent's more adaptable to environments which might break some functionality or require fine-tuning for Opni to work as expected.
```
Comment (kralicky) [inline enhancements/gateway/20230309-customize-capabilty-install.md:4]: grammar: "agent's" -> "agents"

**Review by kralicky — CHANGES_REQUESTED** [review]

File: enhancements/gateway/20230309-customize-capabilty-install.md (hunk 1 above)
Comment (alexandreLamarre) [reply enhancements/gateway/20230309-customize-capabilty-install.md:40]: Might want to explicitly embed CapabilitySpec proto in the `MetricOptions` example input

File: enhancements/gateway/20230309-customize-capabilty-install.md (hunk 2 above)
Comment (joshmeranda) [reply enhancements/gateway/20230309-customize-capabilty-install.md:4]: oops

File: enhancements/gateway/20230309-customize-capabilty-install.md (hunk 1 above)
Comment (joshmeranda) [reply enhancements/gateway/20230309-customize-capabilty-install.md]: updated service name and and included `MetricsCapabilitySpec` into `MetricsOptions`

**Review by alexandreLamarre — APPROVED** [review]

**Review by sanjay920 — APPROVED** [review]

**Pull request merged**